streams/src/main/java/org/apache/kafka/streams/state/internals/LRUCacheEntry.java

```diff
@@ -42,15 +41,17 @@
                   final long timestamp,
                   final int partition,
                   final String topic) {
-        super(timestamp, offset, partition, topic, headers);
-        this.value = value;
+        final ProcessorRecordContext context = new ProcessorRecordContext(timestamp, offset, partition, topic, headers);
+
+        this.record = new ContextualRecord<>(
+            value,
+            context
+        );
+
         this.isDirty = isDirty;
-        this.sizeBytes = (value == null ? 0 : value.length) +
-                1 + // isDirty
-                8 + // timestamp
-                8 + // offset
-                4 + // partition
-                (topic == null ? 0 : topic.length());
+        this.sizeBytes = 1 + // isDirty
+            (value == null ? 0 : value.length) +
+            context.sizeBytes();
```

Comment:
Should we add object overhead for `context` itself? (might be included in `sizeBytes()` if we update is accordingly thought)

Reply:
It would be the responsibility of the context to account for its own overhead, but see my comments above.